criu/files-reg.c

```diff
@@ -1380,6 +1380,14 @@ static bool should_check_size(int flags)
 	return true;
 }
 
+/* Stores the size of a file so that it can be validated while restoring. */
+static inline void store_validation_data_file_size(RegFileEntry *rfe, const struct fd_parms *p)
+{
```

Comment:
To simplify addition of new validation methods, you might want to call this `store_validation_data` from the beginning and update the comment respectively.
If it grows too large we can split it later

Reply:
The code for the build-id method is getting kinda large so I broke everything down into smaller functions preemptively

Reply:
That's ok to have several functions, but there still need to be "the top level validation" function and `dump_one_reg_file` will call it.
And I think having simple checks like file size open coded in that function is ok.

Reply:
@rppt I've made a new commit along with the build-id check, is this fine now?

Reply:
@AnorexicAtticusFinch my point was that from the very beginning you could call the function store_validation_data() and use it for size validation, and then add a function that performs  build-id validation and call it from `store_validation_data()` in the next commit.
Also, I don't think that `_for_restore()` suffix has any useful meaning and IMO can be simply dropped.

Reply:
> @AnorexicAtticusFinch my point was that from the very beginning you could call the function store_validation_data() and use it for size validation, and then add a function that performs build-id validation and call it from `store_validation_data()` in the next commit.
Yeah that's what I've done, I'm calling the build-id functions from store_validation_data and validate_data (store_validation_data_build_id and validate_with_build_id respectively).
> Also, I don't think that `_for_restore()` suffix has any useful meaning and IMO can be simply dropped.
Yeah that makes sense, I'll change the names in the next commit.